### PR TITLE
Added implementing-an-entire-new-exercise.md

### PR DESCRIPTION
**Pull request by acgillette**
File is inside the contributing-to-language-tracks directory.

### CODE REVIEW

**Comment by kytrinyx** [comment]
First off: ✨ 💚 🌺 💚 🌺 💚 ✨ **thank you**

I have feedback, but none of it is major, it's all about smoothing the process out more, and following norms.

One of the most important things when opening a pull request is to provide context about what/why you're fixing. If it's something you stumbled across yourself, then add some background about how and why you think your fix is the right thing. In this case, you're fixing an open issue, so you don't need to convince the maintainers it's worth doing :) They come pre-convinced. That said, there are over 100 maintainers involved in this project, and a bunch of them might not be aware of the open issue. Here your best bet is to add a link to that open issue in the body of the pull request (not the title, but rather in the body of the comment—try editing both to add a link and see why I prefer the body not the title).

Even when providing a link, it's great to summarize the purpose of the PR. Notice that the purpose is just about the only thing a maintainer can't know by looking at it. From looking at the PR even without a description I can quickly see which file you added and where (by clicking on the "files" tab). It's really helpful if you can summarize the _why_ rather than the _what_. Something like "this is part of the effort of consolidating project-wide documentation that is currently scattered across different repositories", or even "this brings in documentation that currently lives in x-common".

The next thing to think about is following the standards of git commits when writing your subject line and body (both for the commits and the PRs). Some people care a lot about this—others not at all. I probably care more than is healthy :) Some people care only about commit messages, others only about PRs. The best short introduction to this is http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html and the best long one is https://chris.beams.io/posts/git-commit/

Finally, if you ended up with an "oops" commit, the maintainers will typically be pretty grateful if you squash your commits into a single commit. If you're newish to Git that can feel ridiculously scary, and of course, Git being Git, there are a bajillion ways you could do it. The good news is, it only takes one! I tend to point people to http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

💛 🌻 💛

**Comment by kytrinyx** [comment]
Oh! One more thing: a bonus trick is to learn how to automatically close the original issue by referencing it with a magic comment in your pull request: https://github.com/blog/1506-closing-issues-via-pull-requests

File: contributing-to-language-tracks/implementing-a-completely-new-exercise.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+A problem must have a unique slug. This slug is used as
+
+* the directory name within each language-specific repository
+* the directory name for the folder that contains the metadata files (in this repository)
+* to identify the exercise in `config.json`
+
+### In exercism/x-common
+
+* Create `exercises/<slug>/description.md` and `exercises/<slug>/metadata.yml`.
+* Bonus: `exercises/<slug>/canonical-data.json` with inputs/outputs for the test suite.
```
Comment (rpottsoh) [inline contributing-to-language-tracks/implementing-a-completely-new-exercise.md]: Question: is the canonical-data.json file really a bonus?  Isn't it really considered more of a requirement when submitting a new exercise to x-common?

**Review by rpottsoh — CHANGES_REQUESTED** [review]
If including a canonical-data.json file is still considered bonus material then ignore my question.

File: contributing-to-language-tracks/implementing-a-completely-new-exercise.md (hunk 1 above)
Comment (kytrinyx) [reply contributing-to-language-tracks/implementing-a-completely-new-exercise.md]: @rpottsoh This change is a simple move from the old location to the new location. We are going to need to overhaul the document in a second round of changes (I think you're right, though that canonical data is probably no longer a bonus).

Comment (ErikSchierboom) [reply contributing-to-language-tracks/implementing-a-completely-new-exercise.md]: I agree. Having canonical data shouldn't really be optional anymore.

**Review by rpottsoh — APPROVED** [review]

**Comment by rpottsoh** [comment]
Went ahead and approved since this is just a move operation on the document.

**Comment by kytrinyx** [comment]
@acgillette let me know how you want to proceed on this one. The options are: merge as is or wait for you to update based on the feedback. 🌷

**Comment by acgillette** [comment]
thank you so much Katrina for the advice! I think it would be best to merge right now, we're about to start capstones and unfortunately I don't think I have the time to work on it more at the moment.

**Comment by kytrinyx** [comment]
That works for me, @acgillette, thank you for your contribution!

**Comment by kytrinyx** [comment]
Merged in b00f220f01e9e644cbf9dc827904a8dd9f126656 ✨ Thank you :heart:

**Comment by kytrinyx** [comment]
(Note: I merged locally)

**Pull request closed without merging**